WcaOnRails/app/controllers/incidents_controller.rb

```diff
@@ -11,13 +11,24 @@ class IncidentsController < ApplicationController
   ]
 
   def index
-    base_model = Incident.includes(:competitions, :incident_tags)
-    if current_user&.can_view_delegate_matters?
-      @incidents = base_model.all
-    else
-      @incidents = base_model.resolved
+    respond_to do |format|
+      format.html {}
+      format.json do
+        @incidents = Incident.all
+        # // todo: filter/search?
+        total_entries = @incidents.length
+        @incidents = @incidents
+                     .page(params[:page] || 1)
+                     .per(params[:entries_per_page] || 10)
+        render json: {
+          totalEntries: total_entries,
+          totalPages: @incidents.total_pages,
```

Comment:
As per my other comment, the Rails `paginate` helper computes these values for you and exposes them, albeit via Response Headers and not within the payload body as you're doing now.

Reply:
These are now accessed via the response headers.